includes/IShader.h

```diff
@@ -30,13 +30,14 @@
 class IShader {
 public:
     virtual void compile() = 0;
-    virtual void checkErrors() = 0;
+    virtual void checkErrors(GLuint &shader) = 0;
```

Comment:
Bonus: Fixed a bug where printing shader errors crashed before the message was printed